components/suggest/src/yelp.rs

```diff
@@ -509,3 +481,238 @@ impl<'a> From<SuggestionBuilder<'a>> for Suggestion {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::{store::tests::TestStore, testing::*, SuggestIngestionConstraints};
+
+    #[test]
+    fn yelp_functions() -> anyhow::Result<()> {
+        before_each();
+
+        let store = TestStore::new(
+            MockRemoteSettingsClient::default()
+                .with_record(SuggestionProvider::Yelp.record("data-4", json!([ramen_yelp()])))
+                .with_record(SuggestionProvider::Yelp.icon(yelp_favicon())),
+        );
+
+        store.ingest(SuggestIngestionConstraints {
+            providers: Some(vec![SuggestionProvider::Yelp]),
+            ..SuggestIngestionConstraints::all_providers()
+        });
+
+        store.read(|dao| {
+            test_find_modifier(dao, "", Modifier::Pre, FindFrom::First, None)?;
```

Comment:
It doesn't look like there are any tests for queries that are substrings of modifiers, like "be", "bes", "chea", etc. Could you add some please, using the different combinations of `Modifier` and `FindFrom`? (In contrast there are such tests for subjects, which is great.)